libraries/provider_firewall_iptables.rb

```diff
@@ -17,123 +17,109 @@
 
-        # prints all the firewall rules
-        # pp new_resource.subresources
-        log_current_iptables
-        if active?
-          Chef::Log.info("#{new_resource} already enabled.")
-        else
-          Chef::Log.debug("#{new_resource} is about to be enabled")
-          shell_out!('iptables -P INPUT DROP')
-          shell_out!('iptables -P OUTPUT DROP')
-          shell_out!('iptables -P FORWARD DROP')
-
-          shell_out!('ip6tables -P INPUT DROP')
-          shell_out!('ip6tables -P OUTPUT DROP')
-          shell_out!('ip6tables -P FORWARD DROP')
-          Chef::Log.info("#{new_resource} enabled.")
-          new_resource.updated_by_last_action(true)
+        %w(iptables ip6tables).each do |svc|
+
+          # must create empty file for service to start
+          file "create empty /etc/sysconfig/#{svc}" do
+            path "/etc/sysconfig/#{svc}"
+            content '# created by chef to allow service to start'
+            not_if { ::File.exists?("/etc/sysconfig/#{svc}") }
+          end
+
+          service svc do
+            action [:enable, :start]
+          end
         end
       end
     end
 
-    def action_disable
-      if active?
-        shell_out!('iptables -P INPUT ACCEPT')
-        shell_out!('iptables -P OUTPUT ACCEPT')
-        shell_out!('iptables -P FORWARD ACCEPT')
-        shell_out!('iptables -F')
-
-        shell_out!('ip6tables -P INPUT ACCEPT')
-        shell_out!('ip6tables -P OUTPUT ACCEPT')
-        shell_out!('ip6tables -P FORWARD ACCEPT')
-        shell_out!('ip6tables -F')
-        Chef::Log.info("#{new_resource} disabled")
-        new_resource.updated_by_last_action(true)
-      else
-        Chef::Log.debug("#{new_resource} already disabled.")
-      end
+    action :restart do
+      next if disabled?(new_resource)
+
+      # prints all the firewall rules
+      log_iptables
+
+      # ensure it's initialized
+      new_resource.rules Hash.new unless new_resource.rules
+      ensure_default_rules_exist(new_resource.rules)
+
+      %w(iptables ip6tables).each do |iptables_type|
+        iptables_filename = "/etc/sysconfig/#{iptables_type}"
+        # ensure a file resource exists with the current iptables rules
+        begin
+          iptables_file = run_context.resource_collection.find(file: iptables_filename)
+        rescue
+          iptables_file = file iptables_filename do
+            action :nothing
+          end
+        end
+        iptables_content = new_resource.rules[iptables_type].sort_by { |k,v| v }.map { |k,v| k }.join("\n")
+        iptables_file.content "#{iptables_content}\n"
+        iptables_file.run_action(:create)
+
+        # if the file was changed, restart iptables
+        if iptables_file.updated_by_last_action?
+          service_affected = service iptables_type do
+            action :nothing
+          end
 
-      service service_name do
-        action [:disable, :stop]
+          new_resource.notifies(:restart, service_affected, :delayed)
```

Comment:
@rdoorn The one issue with reload is that it doesn't clear out any existing rules. So if you remove a chef resource for a firewall rule, and you issue a reload, the rule won't be removed. I suppose we could do a flush with a reload, but at that point, it feels very similar to do restart.

From what I understand, we also risk a kernel panic on devices where the root device was mounted via network if we leave a default-drop in place, flush all the rules, and then issue a reload. (In other words, a brief packet dropping could cause all kinds of issues, which would be worse than briefly opening the firewall when you're changing the ruleset).

Maybe this could be an option?